modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache

```diff
@@ -71,6 +71,10 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
     {{#authMethods}}
             // authentication {{name}} required
             {{#isApiKey}}
+            {{#withAWSV4Signature}}
+            // aws v4 signature authentication required
+            await setAWS4SignatureInterceptor(globalAxios, configuration)
```

Comment:
It looks like this sets the interceptor to the shared axios instance - affecting all requests made with axios in the app
- it's not limited to just endpoints configured for that in the spec
- it adds multiple instances of the interceptor: every time a `someOperationAxiosParamCreator` function is called

I really think the security settings should be applied per endpoint rather than globally

Reply:
What do you think the best way to achieve this? Would it be creating a "shared" axios instance just for the aws4 secured endpoints?

Reply:
1) It was my first thought, too. But I'm concerned it will create much more confusion down the line. Like, users might want to set some app-specific interceptors globally - but some endpoints won't be covered due to a standalone axios instance (please correct me if that's wrong - it's been some time since I actively worked with axios myself)

2) Another option is not setting an interceptor per se (don't call `axios.interceptor.request.use`). An interceptor is essentially a function. You can call it directly inside your operation. This way you can do that granularly, conditionally - only when needed.

3) A variation of the above is setting a custom global interceptor with a condition inside. Individual operations will add some optional flag to request config. The interceptor will analyze this flag and invoke the aws4-axios interceptor or not based on that.

I personally like option 2 best.

Reply:
Option 4:
If the primary use case is adding AWS4 to the whole spec (rather then to select individual operations in it), we don't even need to support it explicitly in the generator. A user can set a global inteceptor and call it a day :)

Reply:
> Option 4:
If the primary use case is adding AWS4 to the whole spec (rather then to select individual operations in it), we don't even need to support it explicitly in the generator. A user can set a global interceptor and call it a day :)

This is exactly my use case and to me it makes sense since an SDK that is built for inter-service communication only defines endpoints that are meant for such scenario and not any other. In my experience also an SDK uses a single type of authentication/authorization type and not multiple ones. Otherwise it's going to be a complex setup to handle. Shall we go with option 4?

Reply:
I mean, in this case perhaps you don't need any change in openapi-generator - you can set the global interceptor directly in your app